skgrf/grf_.pxd

```diff
@@ -153,17 +144,33 @@ cdef extern from "./grf/src/prediction/collector/TreeTraverser.h" namespace "grf
         ) const
 
 
-cdef extern from "./grf/src/prediction/CustomPredictionStrategy.cpp":
+cdef extern from "./grf/src/prediction/DefaultPredictionStrategy.h" namespace "grf":
+    cdef cppclass DefaultPredictionStrategy:
+        DefaultPredictionStrategy() except +
+
+
+cdef extern from "./grf/src/prediction/CausalSurvivalPredictionStrategy.cpp":
```

Comment:
What happens if you comment out these two lines and add the source file [here](https://github.com/crflynn/skgrf/blob/classifier-2.0/build.py#L47) instead, like:

`sources=[path] + ["skgrf/grf/src/prediction/CausalSurvivalPredictionStrategy.cpp"],`

Reply:
This looks like it actually works, although with my lack of C++ knowledge I'm not sure why. I'm guessing this is related to telling the C++ compiler to compile them separately (?) which avoids the collision.

Reply:
With `sources` equal to the `.pyx`, and `cdef extern from "some_file.cpp"; pass` file I think Cython treats all the definitions in that file as one and compiles it as part of the Cython module. I think an alternative is to omit all the `cdef extern from "some_file.cpp"; pass` and include the filenames in `sources`, then they should be compiled separately and linked to the Cython extension.